src/map/entities/battleentity.cpp

```diff
@@ -2691,3 +2691,32 @@ uint16 CBattleEntity::GetBattleTargetID() const
 {
     return m_battleTarget;
 }
+
+bool CBattleEntity::hasEnmity() const
```

Comment:
I know this is going to be very ugly, but to sell how expensive this is (iterating possibly every mob in the zone if you aren't targetted) - can you call this `hasEnmityEXPENSIVE()` to make sure nobody uses it somewhere hot by accident?

Reply:
sure, that works for me. I think the better longer-term fix here is to make a clone of the Notoriety Container but it only contains mobs that have aggroed you through PAI->Engage

Reply:
Yeah, thats the right answer ^

Reply:
The good news is this hasEnmity is sparsely used and only on-demand (trying to mount chocobo, call trust, use a chocobo whistle). All of these are rate limited by the client and/or other systems, so there's little risk of this being overused every tick

Reply:
Updated function name